text/083-boost-stdlib-dev/text.md

```diff
@@ -0,0 +1,786 @@
+    Reals/Ranalysis5.v, Reals/Binomial.v, Reals/Cauchy_prod.v,
+    Reals/Rtrigo_reg.v, Reals/Rseries.v, Reals/AltSeries.v,
+    Reals/Rminmax.v, Reals/Rpow_def.v, Reals/Rlogic.v,
+    Reals/SeqProp.v, setoid_ring/RealField.v, setoid_ring/Rings_R.v,
+    micromega/Lra.v, micromega/Psatz.v, micromega/Fourier_util.v,
+    micromega/Fourier.v, micromega/RMicromega.v, QArith/Qreals.v,
+    Numbers/DecimalR.v, Numbers/HexadecimalR.v</details>
+* coq-roots-nsatz
+  * The Coq standard library, nsatz tactic
+  * Nullstellensatz tactic nsatz. Namespace `Coq.nsatz`.
+  * 2 files, 0.6 kloc
+  * <details><summary>contains:</summary> nsatz/Nsatz.v,
+    nsatz/NsatzTactic.v</details>
+* coq-roots-unicode
+  * The Coq standard library, unicode alternative notations
+  * Unicode alternative notations. Namespace `Coq.Unicode`.
+  * 2 files, 0.1 kloc
+  * <details><summary>contains:</summary> Unicode/Utf8_core.v,
+    Unicode/Utf8.v</details>
+* coq-roots-primitive-int
+  * The Coq standard library, primitive 63 bits integers
+  * Interface for hardware integers (63 rather than 64 bits due to
+    OCaml garbage collector). This enables very efficient arithmetic,
+    for developing tactics for proofs by reflection for
+    instance. Namespace `Coq.Numbers.Cyclic`.
+  * 9 files, 3.8 kloc
+  * <details><summary>contains:</summary>
+    Numbers/Cyclic/Int63/PrimInt63.v, Numbers/Cyclic/Int63/Sint63.v,
+    Numbers/Cyclic/Int63/Cyclic63.v, Numbers/Cyclic/Int63/Uint63.v,
+    Numbers/Cyclic/Int63/Ring63.v, Numbers/Cyclic/Abstract/NZCyclic.v,
+    Numbers/Cyclic/Abstract/DoubleType.v,
+    Numbers/Cyclic/Abstract/CyclicAxioms.v, micromega/ZifyUint63.v,
+    micromega/ZifySint63.v</details>
+* coq-primitive-array
+  * The Coq standard library, primitive persistent arrays
+  * Imperative arrays with a functionnal interface. Enables efficient
+    computations with arrays. Namespace `Coq.Array`.
+  * 2 files, 0.0 kloc
+  * <details><summary>contains:</summary> Array/PArray.v</details>
+* coq-primitive-floats
+  * The Coq standard library, primitive floats
+  * Interface to hardware floats for efficient computations. This
+    offers a basic model of floating-point arithmetic but is not very
+    usable alone. Look at the coq-flocq package for an actual model of
+    floating-point arithmetic, including links to coq-roots-reals and
+    the current coq-primitive-floats. Namespace `Coq.Floats`
+  * 2 files, 0.1 kloc
+  * <details><summary>contains:</summary> Floats/Floats.v, 
+    Floats/FloatLemmas.v</details>
+* coq-roots-string
+  * The Coq standard library, strings
+  * Implementation of string as list of ASCII characters. Namespace
+    `Coq.Strings`.
+  * 9 files, 3.2 kloc
+  * <details><summary>contains:</summary> Strings/ByteVector.v,
+    Strings/HexString.v, Strings/Ascii.v, Strings/String.v,
+    Strings/OctalString.v, Strings/Byte.v, Strings/BinaryString.v,
+    Numbers/DecimalString.v, Numbers/HexadecimalString.v</details>
```

Comment:
No, you could probably try to dive into the history and ask the author of that file why it got put into the `String` directory in the first place.

Reply:
Same reason that ASCII is under strings, byte was originally added just to be efficient strings

Reply:
Hmm, maybe they do belong together in this case. Is there also a vision for Notations to use Gallina code for parsing lists of bytes, or is that intentionally spearate with decimal types and so on?

Reply:
I think right now only string notations parse lists of bytes.  I think there are some vague plans to allow custom entries to specify that strings should be delimited by something other than `"`, which would allow for arbitrary parsing from lists of bytes in custom entries, but I don't think there's currently any plan to move general notation parsing code to be based on lists of bytes / to live in Gallina.